### PR TITLE
tests/uart: Fix failing uart mode tests

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
After reset was updated philip was not being reset after mode change
This commit fixes and helps flush the mode..
Since philip spits out data on reset it also is reset first.

Test on boards that support uart mode.

### CODE REVIEW

File: tests/periph_uart/tests/02__periph_uart_mode.robot [hunk 1]
```diff
@@ -17,8 +18,7 @@ Force Tags          periph  uart
 *** Test Cases ***
 Even Parity 8 Bits
     DUT UART mode should exist  dev=1
-    PHILIP.Setup Uart		parity=${UART_PARITY_EVEN}
-    API Call Should Succeed     Uart Init
```
Comment (smlng) [inline tests/periph_uart/tests/02__periph_uart_mode.robot]: `UART init` is not required?

File: tests/periph_uart/tests/periph_uart_if.py [hunk 2]
```diff
@@ -22,11 +22,16 @@ def uart_init(self, dev=DEFAULT_DEV, baud=DEFAULT_BAUD):
         self.send_cmd("send {} {}".format(dev, "flush"))
         return ret
 
-    def uart_mode(self, data_bits, parity, stop_bits, dev=DEFAULT_DEV):
+    def uart_mode(self, data_bits, parity, stop_bits, dev=DEFAULT_DEV,
+                  baud=DEFAULT_BAUD):
         """Setup databits, parity and stopbits."""
-        return self.send_cmd(
+        self.send_cmd("init {} {}".format(dev, baud))
```
Comment (smlng) [inline tests/periph_uart/tests/periph_uart_if.py]: we should only wrap API calls and not change (or add) functionality

File: tests/periph_uart/tests/01__periph_uart_base.robot [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
 Documentation       Verify basic functionality of the periph UART API.
 
-Suite Setup         Run Keywords    RIOT Reset
-...                                 PHILIP Reset
+Suite Setup         Run Keywords    PHILIP Reset
+...                                 RIOT Reset
+...                                 API Firmware Should Match
+Test Setup          Run Keywords    PHILIP Reset
+...                                 RIOT Reset
 ...                                 API Firmware Should Match
```
[the remark this answers is not in the document]
Comment (MrKevinWeiss) [reply tests/periph_uart/tests/01__periph_uart_base.robot]: Really, we reset the device, one of the reasons was to flush the buffer, we could rename flush buffer maybe?

File: tests/periph_uart/tests/02__periph_uart_mode.robot (hunk 1 above)
Comment (MrKevinWeiss) [reply tests/periph_uart/tests/02__periph_uart_mode.robot]: UART init is required.  Actually it should be called before any mode change which is why I include it in with the uart mode call.  In the current representation it was not called before the second mode change.

File: tests/periph_uart/tests/periph_uart_if.py (hunk 2 above)
Comment (MrKevinWeiss) [reply tests/periph_uart/tests/periph_uart_if.py]: Agreed.

File: tests/periph_uart/tests/01__periph_uart_base.robot (hunk 3 above)
[the remark this answers is not in the document]
Comment (smlng) [reply tests/periph_uart/tests/01__periph_uart_base.robot]: okay is discussed offline: we should add a echo/flush keyword with proper name and documentation to achieve this and not reuse an existing keyword that is misleading

**Comment by MrKevinWeiss** [comment]
So in a future PR we should add a flush buffer command.

File: tests/periph_uart/tests/periph_uart_if.py (hunk 2 above)
Comment (MrKevinWeiss) [reply tests/periph_uart/tests/periph_uart_if.py]: Hmm many I can change the command to `uart_init_mode`?  Again this should happen every time.  The thing is many boards allow uart mode to be changed and not needing to reinit.  I think it is just the STM that has this constraint... (the registers are only or'ed not cleared with an and then or'ed)

Comment (smlng) [reply tests/periph_uart/tests/periph_uart_if.py]: sounds like this should be fixed in RIOT in the cpu specific uart_mode implementation?

Comment (smlng) [reply tests/periph_uart/tests/periph_uart_if.py]: okay as a quick fix lets have a separate call `uart_init_mode` in the python interface that does that, i.e. calling `mode` and `init` afterwards. but do not change (or extend) the existing `init` and `mode` wrappers, okay?

**Comment by MrKevinWeiss** [comment]
I see what you mean by wrapper now.  How is this?

**Comment by MrKevinWeiss** [comment]
Closing in favor of #17

**Pull request closed without merging**